### ISSUE TITLE
ITelnetTransport API documentation is incorrect for refusal case

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4677]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4677|
|Type|defect|
|Created|2010-10-07 22:31:05Z|
|Branch|https://github.com/twisted/twisted/tree/telnet-transport-docs-4677-2|

The `do` and `will` methods claim that the `Deferred` will fire with `False` if the option is refused.  In fact, in the only implementation, the `do` Deferred will fire with `OptionRefused` and the `will` Deferred will probably errback with an `AssertionError` because of [#4231](https://github.com/twisted/twisted/issues/4231).

The docs should be updated to reflect the implementation (not the reverse, since people might actually have depended on the currently implemented behavior).

<details><summary>Searchable metadata</summary>

```
trac-id__4677 4677
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_telnet_transport_docs_4677_2 branches/telnet-transport-docs-4677-2
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__documentation documentation
time__1286490665000000 1286490665000000
changetime__1287410976000000 1287410976000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [30111]) Branching to 'telnet-transport-docs-4677'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30119]) Branching to 'telnet-transport-docs-4677-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Simple doc changes in the branch, please review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [30142]) Merge telnet-transport-docs-4677-2

Author: exarkun
Reviewer: therve
Fixes: [#4677](https://github.com/twisted/twisted/issues/4677)

Document a state tracking class internal to `twisted.conch.telnet` used for tracking
the progress and results of telnet option negotiation.  Also correct the interface
documentation on `ITelnetTransport` to correctly describe the behavior in the case
an option is refused.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
